AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ExtraHandler.java

```diff
@@ -72,12 +89,34 @@ public void process(Element element, HasExtras holder) {
 		}
 
 		JFieldVar extraKeyStaticField = createStaticExtraField(holder, extraKey, fieldName);
-		injectExtraInComponent(element, holder, extraKeyStaticField, fieldName);
-
 		if (holder instanceof HasIntentBuilder) {
 			String docComment = getProcessingEnvironment().getElementUtils().getDocComment(element);
-			createIntentInjectionMethod(element, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
+			createIntentInjectionMethod(param, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
```

Comment:
Hmm. Maybe you can expose some internals then from injecthelper, which could be used by the handler.

The injecthelper could mark the method parameters which were already handled. Then the handler could query this information, and the the same logic i wrote down.